### PR TITLE
Journalist must sign in to be able to access create article functionality

### PR DESCRIPTION
**Pull request by lukeperrin10**
pt story: https://www.pivotaltracker.com/story/show/176030917

## User Story 

```
As an journalist,
In order to be able to write articles
I would like to be able to access an article creation form on my dashboard when authenticated
```

## Suggested Changes 

- Adds secure login page to site, to protect article creation 

## Added dependencies 
- React Router

## What we have learned 
- How to redirect on authentication 
_________
image

### CODE REVIEW

File: cypress/fixtures/handleLogin.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "status": "success",
+  "data": {
+    "id": 1,
+    "email": "user@mail.com",
+    "provider": "email",
+    "uid": "user@mail.com",
+    "allow_password_change": false
+  }
+}
```
Comment (LucasKnudsen) [inline cypress/fixtures/handleLogin.json]: There's data sent from the API not reflected here. You should talk to the API team 😏

File: cypress/integration/userCanSeeLogInLandingPage.feature.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      cy.get('[data-cy=login-form]').should('exist');
+      cy.get('[data-cy=login-form]').within(() => {
+        cy.get('[data-cy=login-username]').type('user@mail.com');
+        cy.get('[data-cy=login-password]').type('password');
+        cy.get('[data-cy=login-btn]').click();
+      });
+      cy.get('[data-cy=create-article-btn]').should('be.visible');
+    });
+  });
+
```
Comment (LucasKnudsen) [inline cypress/integration/userCanSeeLogInLandingPage.feature.js:27]: Could we display the name of the user?

File: src/modules/Credentials.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import store from '../state/store/configureStore';
+
+const Credentials = {
+  getFormInput(event) {
+    return {
+      email: event.target.username.value,
+      password: event.target.password.value,
+    };
+  },
```
Comment (LucasKnudsen) [inline src/modules/Credentials.js]: Should include `source: 'admin-system` :D

**Comment by LucasKnudsen** [comment]
And check the pull request in Semaphore settings 🚀

File: src/modules/Credentials.js (hunk 3 above)
Comment (offcast-dev) [reply src/modules/Credentials.js]: done

File: cypress/integration/userCanSeeLogInLandingPage.feature.js (hunk 2 above)
Comment (offcast-dev) [reply cypress/integration/userCanSeeLogInLandingPage.feature.js:27]: It is not really in scope now, so we will do it in the next PR

File: src/index.css [hunk 4]
```diff
@@ -0,0 +1,4 @@
+body {
```
Comment (1CIM) [inline src/index.css]: Since u use inline style is it necessary to use a css file for only background color?

File: src/App.jsx [hunk 5]
```diff
@@ -1,10 +1,17 @@
...
+import PopupMessage from './components/PopupMessage'
+import { Switch, Route } from 'react-router';
 
 const App = () => {
   return (
     <>
-      <Dashboard />
+    <Switch>
+      <Route exact path="/" component={LogInLandingpage}></Route>
+      <Route exact path="/dashboard" component={Dashboard}></Route>
```
Comment (1CIM) [inline src/App.jsx]: Double quotes needed? can't use single quotes for path?

File: cypress/fixtures/handleLogin.json (hunk 1 above)
Comment (offcast-dev) [reply cypress/fixtures/handleLogin.json]: fixed

File: src/index.css (hunk 4 above)
Comment (offcast-dev) [reply src/index.css]: it is the same background everywhere, I think it will be populated with more stuff later

Comment (offcast-dev) [reply src/index.css]: changed to be in App.jsx, index.css removed

**Review by LucasKnudsen — APPROVED** [review]

**Review by tochman — APPROVED** [review]
Looks good now.

**Pull request merged**